tests/test_locality_NeighborQuery.py

```diff
@@ -512,39 +508,40 @@ def test_corner_2d(self):
         result = list(nq.query(positions[[0]], dict(mode="nearest", num_neighbors=3)))
         assert get_point_neighbors(result, 0) == {0, 1, 2}
 
-    def test_random_system_query(self):
+    inputs = {10: [1, 5], 100: [1, 5, 10, 50], 500: [1, 5, 10, 50]}
+
+    @pytest.mark.parametrize(
+        "N, k", [(N, k) for N, value in inputs.items() for k in value]
+    )
+    def test_random_system_query(self, N, k):
         np.random.seed(0)
         L = 10
         box = freud.box.Box.cube(L)
 
         # Generate random points
-        for N in [10, 100, 500]:
-            positions = box.wrap(L / 2 * np.random.rand(N, 3))
-            ks = [1, 5]
-            if N > 10:
-                ks.extend([10, 50])
-            for k in ks:
-                nq = self.build_query_object(box, positions, L / 10)
-
-                nlist = nq.query(
-                    positions, dict(num_neighbors=k, exclude_ii=True)
-                ).toNeighborList()
-                assert len(nlist) == k * N, (
-                    "Wrong nlist length for N = {}," "num_neighbors = {}, length = {}"
-                ).format(N, k, len(nlist))
-                nlist_array = nlist[:]
-                for i in range(N):
-                    assert not ([i, i] == nlist_array).all(axis=1).any()
-
-                nlist = nq.query(
-                    positions, dict(num_neighbors=k, exclude_ii=False)
-                ).toNeighborList()
-                assert len(nlist) == k * N, (
-                    "Wrong nlist length for N = {}, " "num_neighbors = {}, length = {}"
-                ).format(N, k, len(nlist))
-                nlist_array = nlist[:]
-                for i in range(N):
-                    assert ([i, i] == nlist_array).all(axis=1).any()
+        positions = box.wrap(L / 2 * np.random.rand(N, 3))
+
+        nq = self.build_query_object(box, positions, L / 10)
+
+        nlist = nq.query(
+            positions, dict(num_neighbors=k, exclude_ii=True)
+        ).toNeighborList()
+        assert len(nlist) == k * N, (
+            "Wrong nlist length for N = {}," "num_neighbors = {}, length = {}"
+        ).format(N, k, len(nlist))
```

Comment:
This falls into the same category of assertion error messages that are no longer needed when using `parametrize`. Same for a few other assertions in this test function and perhaps elsewhere.

```suggestion
        assert len(nlist) == k * N
```